engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -1120,18 +1131,26 @@ private AgentAttache handleConnectedAgent(final Link link, final StartupCommand[
 
             final HostVO host = _resourceMgr.createHostVOForConnectedAgent(startup);
             if (host != null) {
-                ready = new ReadyCommand(host.getDataCenterId(), host.getId(), NumbersUtil.enableHumanReadableSizes);
-
-                if (!indirectAgentLB.compareManagementServerList(host.getId(), host.getDataCenterId(), agentMSHostList, lbAlgorithm)) {
-                    final List<String> newMSList = indirectAgentLB.getManagementServerList(host.getId(), host.getDataCenterId(), null);
-                    ready.setMsHostList(newMSList);
-                    ready.setLbAlgorithm(indirectAgentLB.getLBAlgorithmName());
-                    ready.setLbCheckInterval(indirectAgentLB.getLBPreferredHostCheckInterval(host.getClusterId()));
-                    s_logger.debug("Agent's management server host list is not up to date, sending list update:" + newMSList);
-                }
+                GlobalLock joinLock = getHostJoinLock(host.getId());
```

Comment:
I am not sure what you mean by life-lock.

If you mean a deadlock, as per my understanding of Agent's implementation, it will first send a disconnect (not waiting for disconnect to complete on server) and then send a connect to a different server. So, it's unlikely. Even if that happens, it will wait for lock for only 60 seconds.

Reply:
no @vishesh92, I did mean a life-lock. a dead-lock means it waits for a lock that won´t be released, A life lock means two parties keep trying but keep hitting each others lock. In this case a disconnect hold the lock while the connect on the other side tries for it and fails as the disconnect is busy (for just to long) and than the other way around.

Reply:
@DaanHoogland No. It won't happen. We disconnect only when the connection was successful with the management server.